R/uniDimBayesianOutputTables.R

```diff
@@ -0,0 +1,239 @@
+  scaleTable <- createJaspTable(gettext("Bayesian Scale Reliability Statistics"))
+  scaleTable$dependOn(options = c("credibleIntervalValueScale","meanScale", "sdScale", "rHat",
+                                  "alphaScale", "omegaScale", "lambda2Scale", "lambda6Scale", "glbScale",
+                                  "averageInterItemCor", "meanMethod", "sdMethod"))
+
+  scaleTable$addColumnInfo(name = "estimate", title = gettext("Estimate"), type = "string")
+  intervalLow <- gettextf("%s%% CI",
+                          format(100*options[["credibleIntervalValueScale"]], digits = 3, drop0trailing = TRUE))
+  intervalUp <- gettextf("%s%% CI",
+                         format(100*options[["credibleIntervalValueScale"]], digits = 3, drop0trailing = TRUE))
+  intervalLow <- gettextf("%s lower bound", intervalLow)
+  intervalUp <- gettextf("%s upper bound", intervalUp)
+
+  if (options[["rHat"]]) {
+    allData <- data.frame(estimate = c("Posterior mean", intervalLow, intervalUp, "R-hat"))
+  } else {
+    allData <- data.frame(estimate = c("Posterior mean", intervalLow, intervalUp))
+  }
+
+  derivedOptions <- model[["derivedOptions"]]
+  opts     <- derivedOptions[["namesEstimators"]][["tables"]]
+  selected <- derivedOptions[["selectedEstimators"]]
+  idxSelected <- which(selected)
+  # match names from model object with the names from selected coefficients to find their index:
+  idMatchedNames <- which(!is.na(charmatch(names(model), names(selected[idxSelected]))))
+
+  if (!is.null(model[["empty"]])) {
+    scaleTable$setData(allData)
+    nvar <- length(options[["variables"]])
+    if (sum(selected) > 0L && nvar < 3) {
+      for (i in idxSelected) {
+        scaleTable$addColumnInfo(name = paste0("est", i), title = opts[i], type = "number")
+      }
+    }
+    scaleTable$addFootnote(model[["footnote"]])
+    scaleTable$position <- 1
+    stateContainerB <- .getStateContainerB(jaspResults)
+    stateContainerB[["scaleTable"]] <- scaleTable
+    return()
+  }
+
+  z <- 1
+  for (i in idxSelected) {
+
+    scaleTable$addColumnInfo(name = paste0("est", i), title = opts[i], type = "number")
+
+    if (options[["rHat"]]) {
+      if (opts[i] == "mean" || opts[i] == "sd") {
+        rhat <- NA_real_
+      } else {
+        tmp <- lapply(as.data.frame(t(model[[idMatchedNames[z]]][["samp"]])), coda::mcmc)
+        rhat <- coda::gelman.diag(coda::as.mcmc.list(tmp))[["psrf"]][, 1]
+      }
+      newData <- data.frame(est = c(unlist(model[[idMatchedNames[z]]][["est"]], use.names = F),
+                                    unlist(model[[idMatchedNames[z]]][["cred"]], use.names = F),
+                                    rhat))
+    } else {
+      newData <- data.frame(est = c(unlist(model[[idMatchedNames[z]]][["est"]], use.names = F),
+                                    unlist(model[[idMatchedNames[z]]][["cred"]], use.names = F)))
+    }
+    colnames(newData) <- paste0(colnames(newData), i)
+    allData <- cbind(allData, newData)
+    z <- z+1
+  }
```

Comment:
```suggestion
  if (!is.null(model[["empty"]])) {
    scaleTable$setData(allData)
    nvar <- length(options[["variables"]])
    if (sum(selected) > 0L && nvar < 3) {
      for (i in idxSelected) {
        scaleTable$addColumnInfo(name = paste0("est", i), title = opts[i], type = "number")
      }
    }
    scaleTable$addFootnote(model[["footnote"]])
    scaleTable$position <- 1
    stateContainerB <- .getStateContainerB(jaspResults)
    stateContainerB[["scaleTable"]] <- scaleTable
    return()
  }

  for (j in seq_along(idxSelected)) {

    i <- idxSelected[j]
    nm <- names(idxSelected[j])

    scaleTable$addColumnInfo(name = paste0("est", i), title = opts[i], type = "number")

    newData <- data.frame(est = c(unlist(model[[nm]][["est"]], use.names = FALSE),
                                  unlist(model[[nm]][["cred"]], use.names = FALSE)))
    if (options[["rHat"]]) {
      if (opts[i] == "mean" || opts[i] == "sd") {
        rhat <- NA_real_
      } else {
        tmp <- lapply(as.data.frame(t(model[[nm]][["samp"]])), coda::mcmc)
        rhat <- coda::gelman.diag(coda::as.mcmc.list(tmp))[["psrf"]][, 1]
      }
      newData[["rhat"]] <- rhat
    }

    colnames(newData) <- paste0(colnames(newData), i)
    allData <- cbind(allData, newData)
  }
```

Reply:
ah, ja. That does indeed seem easier. Completely slipped my mind